### PR TITLE
Allow identification of separate workers for blueprint debugging

### PR DESCRIPTION
**Pull request by firtina-improbable**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
See https://github.com/improbableio/UnrealEngine/pull/129

#### Release note
- If using Spatial Networking, the blueprint world selector dropdown now shows the worker ID for each world for easier debugging.

#### Tests
Tested locally by launching multiple workers, selecting a blueprint that's active in all worlds and checking what's displayed in the "world selection" dropdown.

image

#### Documentation
N/A

### CODE REVIEW

**Comment by improbable-valentyn** [comment]
Increment the version in `EngineVersionCheck.h` (if that versioning was merged into hydra)

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 1]
```diff
@@ -524,6 +524,11 @@ void USpatialNetDriver::NotifyActorDestroyed(AActor* ThisActor, bool IsSeamlessT
 	RenamedStartupActors.Remove(ThisActor->GetFName());
 }
 
+const FString USpatialNetDriver::GetUniqueIdentifier() const
+{
+	return Connection->GetWorkerId();
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: Ensure Connection is valid before using.

Comment (firtina-improbable) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: Added check above

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 2]
```diff
@@ -524,6 +524,14 @@ void USpatialNetDriver::NotifyActorDestroyed(AActor* ThisActor, bool IsSeamlessT
 	RenamedStartupActors.Remove(ThisActor->GetFName());
 }
 
+const FString USpatialNetDriver::GetUniqueIdentifier() const
+{
+	if (Connection == nullptr)
```
Comment (danielimprobable) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: Wrap in {}

File: SpatialGDK/Source/SpatialGDK/Public/Utils/EngineVersionCheck.h [hunk 3]
```diff
@@ -7,7 +7,7 @@
 // GDK Version to be updated with SPATIAL_ENGINE_VERSION
 // when breaking changes are made to the engine that requires
 // changes to the GDK to remain compatible
-#define SPATIAL_GDK_VERSION 1
+#define SPATIAL_GDK_VERSION 2
```
Comment (danielimprobable) [inline SpatialGDK/Source/SpatialGDK/Public/Utils/EngineVersionCheck.h]: Why is this only two, surely we've made a lot more versions than that, is this still used?

@m-samiec

Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Public/Utils/EngineVersionCheck.h]: It's only recently been introduced.

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 1 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: How about -
if (Connection != nullptr)
{
return etc.
}
return Super::GetUniqueIdentified();

Comment (firtina-improbable) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: Aye sounds better

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 4]
```diff
@@ -524,6 +524,11 @@ void USpatialNetDriver::NotifyActorDestroyed(AActor* ThisActor, bool IsSeamlessT
 	RenamedStartupActors.Remove(ThisActor->GetFName());
 }
 
+const FString USpatialNetDriver::GetUniqueIdentifier() const
+{
+	return Connection->GetWorkerId();
+}
+
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: Might we want to return `Connection->GetWorkerId() + Super::GetUniqueIdentifier()` ?

**Comment by firtina-improbable** [comment]
Linking ticket UNR-1384

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 4 above)
Comment (firtina-improbable) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: Currently super is `""` - if we are in the future going to add a different unique identifier or change the class ancestry, any problems or further work should be trivial to reconsider at that point.

File: ci/unreal-engine.version [hunk 5]
```diff
@@ -1 +1 @@
-UnrealEngine-24edbad2f25c822bb8ec62846ce80bc1e41d1bdf
+UnrealEngine-1745cc0e1bed2c2126e46f5f3bd4a45515a6d6bf
```
Comment (m-samiec) [inline ci/unreal-engine.version]: There's a new line at the end of this line. That's why BK is failing.

**Pull request merged**